### ISSUE TITLE
Record structure does not contain field "is_configurable": 

### ISSUE THREAD

**Issue by lokisp** [magento/data-migration-tool#595]
I have problem when migration data from magento1 to magento2. 
It show Record structure does not contain field "is_configurable": 
### Preconditions
1.  Magento 1.9.2.4 to Magento 2.2.6
2. Error:
https://prnt.sc/kzgbj8

### Steps to reproduce

1.  Setup new magento 2.2.6
2. Install migration data tool
3. Setup config and map field
4. Run command settings and migration

### Expected result

1. No problem with it 

### Actual result
https://prnt.sc/kzgbj8
It shows : Record structure does not contain field "is_configurable"
Actually, I checked database of magento1. It is default column magento.
However, I add additional ignore in file map-eav.xml:
      `<ignore>
            <field>eav_attribute.is_configurable</field>
        </ignore>`

My config file:

> <config xmlns:xs="http://www.w3.org/2001/XMLSchema-instance" xs:noNamespaceSchemaLocation="../../config.xsd">
    <steps mode="settings">
        <step title="Settings Step">
            <integrity>Migration\Step\Settings\Integrity</integrity>
            <data>Migration\Step\Settings\Data</data>
        </step>
        <step title="Stores Step">
            <integrity>Migration\Step\Stores\Integrity</integrity>
            <data>Migration\Step\Stores\Data</data>
            <volume>Migration\Step\Stores\Volume</volume>
        </step>
    </steps>
    <steps mode="data">
        <step title="Data Integrity Step">
            <integrity>Migration\Step\DataIntegrity\Integrity</integrity>
        </step>
        <step title="EAV Step">
            <integrity>Migration\Step\Eav\Integrity</integrity>
            <data>Migration\Step\Eav\Data</data>
            <volume>Migration\Step\Eav\Volume</volume>
        </step>
        <step title="Customer Attributes Step">
            <integrity>Migration\Step\Customer\Integrity</integrity>
            <data>Migration\Step\Customer\Data</data>
            <volume>Migration\Step\Customer\Volume</volume>
        </step>
        <step title="Map Step">
            <integrity>Migration\Step\Map\Integrity</integrity>
            <data>Migration\Step\Map\Data</data>
            <volume>Migration\Step\Map\Volume</volume>
        </step>
        <step title="Url Rewrite Step">
            <integrity>Migration\Step\UrlRewrite\Version191to2000</integrity>
            <data>Migration\Step\UrlRewrite\Version191to2000</data>
            <volume>Migration\Step\UrlRewrite\Version191to2000</volume>
        </step>
        <step title="Log Step">
            <integrity>Migration\Step\Log\Integrity</integrity>
            <data>Migration\Step\Log\Data</data>
            <volume>Migration\Step\Log\Volume</volume>
        </step>
        <step title="Ratings Step">
            <integrity>Migration\Step\Ratings\Integrity</integrity>
            <data>Migration\Step\Ratings\Data</data>
            <volume>Migration\Step\Ratings\Volume</volume>
        </step>
        <step title="ConfigurablePrices step">
            <integrity>Migration\Step\ConfigurablePrices\Integrity</integrity>
            <data>Migration\Step\ConfigurablePrices\Data</data>
            <volume>Migration\Step\ConfigurablePrices\Volume</volume>
        </step>
        <step title="OrderGrids Step">
            <integrity>Migration\Step\OrderGrids\Integrity</integrity>
            <data>Migration\Step\OrderGrids\Data</data>
            <volume>Migration\Step\OrderGrids\Volume</volume>
        </step>
        <step title="Tier Price Step">
            <integrity>Migration\Step\TierPrice\Integrity</integrity>
            <data>Migration\Step\TierPrice\Data</data>
            <volume>Migration\Step\TierPrice\Volume</volume>
        </step>
        <step title="SalesIncrement Step">
            <integrity>Migration\Step\SalesIncrement\Integrity</integrity>
            <data>Migration\Step\SalesIncrement\Data</data>
            <volume>Migration\Step\SalesIncrement\Volume</volume>
        </step>
        <step title="PostProcessing Step">
            <data>Migration\Step\PostProcessing\Data</data>
        </step>
    </steps>
    <steps mode="delta">
        <step title="Customer Attributes Step">
            <delta>Migration\Step\Customer\Delta</delta>
            <volume>Migration\Step\Customer\Volume</volume>
        </step>
        <step title="Map Step">
            <delta>Migration\Step\Map\Delta</delta>
            <volume>Migration\Step\Map\Volume</volume>
        </step>
        <step title="Log Step">
            <delta>Migration\Step\Log\Delta</delta>
            <volume>Migration\Step\Log\Volume</volume>
        </step>
        <step title="OrderGrids Step">
            <delta>Migration\Step\OrderGrids\Delta</delta>
            <volume>Migration\Step\OrderGrids\Volume</volume>
        </step>
        <step title="SalesIncrement Step">
            <delta>Migration\Step\SalesIncrement\Delta</delta>
            <volume>Migration\Step\SalesIncrement\Volume</volume>
        </step>
    </steps>
    <source>
        <database host="localhost" name="xxxx" user="root" password="xxxx"/>
    </source>
    <destination>
        <database host="localhost" name="xxxxxxx" user="root" password="xxxxx"/>
    </destination>
    <options>
        <map_file>etc/opensource-to-opensource/1.9.2.4/map.xml</map_file>
        <eav_map_file>etc/opensource-to-opensource/map-eav.xml</eav_map_file>
        <eav_document_groups_file>etc/opensource-to-opensource/eav-document-groups.xml</eav_document_groups_file>
        <eav_attribute_groups_file>etc/opensource-to-opensource/eav-attribute-groups.xml</eav_attribute_groups_file>
        <log_map_file>etc/opensource-to-opensource/map-log.xml</log_map_file>
        <log_document_groups_file>etc/opensource-to-opensource/log-document-groups.xml</log_document_groups_file>
        <settings_map_file>etc/opensource-to-opensource/settings.xml</settings_map_file>
        <customer_map_file>etc/opensource-to-opensource/map-customer.xml</customer_map_file>
        <customer_document_groups_file>etc/opensource-to-opensource/customer-document-groups.xml</customer_document_groups_file>
        <customer_attribute_groups_file>etc/opensource-to-opensource/customer-attribute-groups.xml</customer_attribute_groups_file>
        <delta_document_groups_file>etc/opensource-to-opensource/deltalog.xml</delta_document_groups_file>
        <order_grids_document_groups_file>etc/opensource-to-opensource/order-grids-document-groups.xml</order_grids_document_groups_file>
        <map_document_groups>etc/opensource-to-opensource/map-document-groups.xml</map_document_groups>
        <class_map>etc/opensource-to-opensource/class-map.xml</class_map>
        <tier_price_map_file>etc/opensource-to-opensource/map-tier-price.xml</tier_price_map_file>
        <stores_map_file>etc/opensource-to-opensource/map-stores.xml</stores_map_file>
        
        <bulk_size>0</bulk_size>
        
        <direct_document_copy>0</direct_document_copy>
        <source_prefix />
        <dest_prefix />
        <auto_resolve_urlrewrite_duplicates>0</auto_resolve_urlrewrite_duplicates>
        <log_file>migration.log</log_file>
        <progress_bar_format>%percent%% [%bar%] Remaining Time: %remaining%</progress_bar_format>
        <upgrade_customer_password_hash>1</upgrade_customer_password_hash>
        <edition_migrate>opensource-to-opensource</edition_migrate>
        <edition_number>1.9.2.4</edition_number>
        <init_statements_source>SET NAMES utf8;</init_statements_source>
        <init_statements_destination>SET NAMES utf8;</init_statements_destination>
        <crypt_key>ee8c2f57c8e64cc641ed58cccf7b31eb</crypt_key>
    </options>
</config>

However I still meet same problem. If I run again. It will show default attribute exist (I must reinstall and try again but nothing help)

**Comment by victor-v-rad** [comment]
Hi @lokisp 

The error is thrown by Map step. Check your map.xml file contain
```
            <ignore>
                <document>catalog_eav_attribute</document>
            </ignore>
```

**Comment by dsmithhayes** [comment]
I am getting an identical error except I am migration from Enterprise 1.14.1.0 to Enterprise 2.2.6. However the field that is not apart of the Record Structure is the `font_name`. 

Edit, I have found a column in `catalog_product_option_type_value` table called `font_name`. I believe this is the offending column.

**Comment by victor-v-rad** [comment]
Hi @lokisp @dsmithhayes 
Internal ticket MAGETWO-95455. Please try this fix 13728b3

**Comment by lokisp** [comment]
Hi Victor,
I wonder about map.xml . It ignore many table. May I know the table should migration? example eav table. Thank you.

**Comment by dsmithhayes** [comment]
@victor-v-rad 

Thank you. I have noticed that the fields in my Magento 1 (Enterprise 1.14.1.0) are the result of custom modules appending columns to tables. I have noted other tables with issues in regards to non-matching fields:

`catalog_product_option_type_value`
`newsletter_subscriber`
`sales_flat_creditmemo`
`sales_flat_creditmemo_item`
`sales_flat_invoice`
`sales_flat_invoice_item`
`sales_flat_order`
`sales_flat_order_item`

Common characteristics between each un-mappable field is they are not nullable and from third party sources. The list of fields is long and relevent to our third party extensions but I am willing to provide that information with you.

I would also like to note that The Data Migration Tool versioned `2.2.5` would successfully ignore these columns by default.

**Comment by victor-v-rad** [comment]
@lokisp eav tables migrate in EAV Step. To have 'Record structure does not contain field' fixed use `2.2.5` ver of the tool or this fix 13728b3

**Comment by tuyennn** [comment]
Confirmed 13728b3 fixed the problem on 2.2.6 also
